File: RecyclerViewJavaLessons(17)/RecyclerViewCustomAdapterLoadingMore_4/app/src/main/java/com/example/recyclerviewcustomadapterloadingmore/activity/MainActivity.java
package com.example.recyclerviewcustomadapterloadingmore.activity;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Context;
import android.os.Bundle;
import android.util.Log;
import android.widget.Toast;

import com.example.recyclerviewcustomadapterloadingmore.Member.Member;
import com.example.recyclerviewcustomadapterloadingmore.R;
import com.example.recyclerviewcustomadapterloadingmore.adapter.CustomAdapter;
import com.example.recyclerviewcustomadapterloadingmore.listener.OnBottomReachedListener;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    private Context context;
    private RecyclerView recyclerView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initViews();

        List<Member> members = prepareMemberList();
        refreshAdapter(members);

    }

    private void initViews() {
        recyclerView = findViewById(R.id.recyclerView);
        recyclerView.setLayoutManager(new GridLayoutManager(context,1));
    }

    private void refreshAdapter(List<Member> members){
       CustomAdapter adapter = new CustomAdapter(context,members, new OnBottomReachedListener() {
           @Override
           public void onBottomReached(int position) {
//               Log.d("@@@", "@@@position" + position);
      //         Toast.makeText(context, "Loading More..." + position, Toast.LENGTH_SHORT).show();
           }
       });
       recyclerView.setAdapter(adapter);
    }

    List<Member> prepareMemberList() {
        List<Member> members = new ArrayList<>();
        members.add(new Member());
        for (int i = 0; i < 20; i++) {
            if (i==0 || i==5 || i == 10 || i == 15){
                members.add(new Member("Feruz", "Davlatov",false));
            } else {
                members.add(new Member("Feruz", "Davlatov",true));
            }
        }
        members.add(new Member());
        return members;
    }

}

File: RecyclerViewKotlinLessons(17)/RecyclerViewCustomBasicAdapter_Kotlin_1/app/src/main/java/com/example/recyclerviewcustombasicadapter_kotlin_1/MainActivity.kt
package com.example.recyclerviewcustombasicadapter_kotlin_1


import android.content.Context
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import java.util.ArrayList



class MainActivity : AppCompatActivity() {
    private lateinit var recyclerView: RecyclerView
    private lateinit var context: Context

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initViews()

        val member = prepareMemberList()
        refreshAdapter(member)

    }



    private fun initViews() {
        context = this
        recyclerView = findViewById(R.id.recyclerView)
        recyclerView.layoutManager = GridLayoutManager(context, 1)
    }

    private fun refreshAdapter(members: List<Member>) {
        val adapter = CustomAdapter(context, members)
        recyclerView.adapter = adapter
    }


    fun prepareMemberList(): List<Member> {
        val members: MutableList<Member> = ArrayList()
        for (i in 0..19) {
            if (i == 0 || i == 5 || i == 10 || i == 15) {
                members.add(Member("Feruz", "Davlatov",true))
            } else {
                members.add(Member("Feruz", "Davlatov",false))
            }
        }
        return members
    }

}




